sentry-android-core/src/main/java/io/sentry/android/core/InternalSentrySdk.java

```diff
@@ -252,6 +263,26 @@ private static void addTimeSpanToSerializedSpans(TimeSpan span, List<Map<String,
     spans.add(spanMap);
   }
 
+  private static void deleteCurrentSessionFile(final @NotNull SentryOptions options) {
+    final String cacheDirPath = options.getCacheDirPath();
+    if (cacheDirPath == null) {
+      options.getLogger().log(INFO, "Cache dir is not set, not deleting the current session.");
+      return;
+    }
+
+    if (!options.isEnableAutoSessionTracking()) {
+      options
+          .getLogger()
+          .log(DEBUG, "Session tracking is disabled, bailing from deleting current session file.");
+      return;
+    }
+
+    final File sessionFile = EnvelopeCache.getCurrentSessionFile(cacheDirPath);
+    if (!sessionFile.delete()) {
```

Comment:
is this happening in the main thread?

Reply:
depends on the hybrid platform that will be used, on RN not, on Flutter yes

Reply:
For Flutter I guess because the invocation of the method channels happen on the main thread?